extensions/extension.go

```diff
@@ -158,3 +173,42 @@ func evalFunc(env any, name string, args []object.Object) object.Object {
 	}
 	return res
 }
+
+func saveFunc(env any, _ string, args []object.Object) object.Object {
```

Comment:
Does this method must comply with an interface?

You should add a comment then, because the unused string is awkward

Reply:
yeah it's defined a bit above

```go
	jsonFn.Callback = saveFunc // save to file.
```
which is
```go
// ExtFunction is the signature of what grol will call when the extension is invoked.
// Incoming arguments are validated for type and number of arguments based on [Extension].
// eval is the opaque state passed from the interpreter, it can be used with eval.Eval etc
// name is the function name as registered under.
type ExtFunction func(eval any, name string, args []Object) Object
```

like everything in the file, in short it doesn't need to know its own name